File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/ScriptOutputObject.java
package com.github.dedis.popstellar.model.objects;

public class ScriptOutputObject {

  // Type of script
  private final String type;
  // Hash of the recipient’s public key
  private final String pubKeyHash;

  /**
   * @param type Type of script
   * @param pubKeyHash Hash of the recipient’s public key
   */
  public ScriptOutputObject(String type, String pubKeyHash) {
    this.type = type;
    this.pubKeyHash = pubKeyHash;
  }

  public String getType() {
    return type;
  }

  public String getPubKeyHash() {
    return pubKeyHash;
  }

  // a function that given a public key verify it is the goof public key

}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/digitalcash/ScriptOutputObject.kt
package com.github.dedis.popstellar.model.objects.digitalcash

import com.github.dedis.popstellar.model.Immutable

@Immutable
class ScriptOutputObject
/**
 * @param type Type of script
 * @param pubKeyHash Hash of the recipient’s public key
 */
( // Type of script
    val type: String, // a function that given a public key verify it is the goof public key
    // Hash of the recipient’s public key
    val pubKeyHash: String
)
